apps/web/src/actions/documents/create.ts

```diff
@@ -15,6 +15,13 @@ export const createDocumentVersionAction = withProject
     { type: 'json' },
   )
   .handler(async ({ input }) => {
-    const result = await createDocumentVersion(input)
+    const commit = await findCommitByUuid({
+      projectId: input.projectId,
+      uuid: input.commitUuid,
+    }).then((r) => r.unwrap())
+    const result = await createNewDocument({
+      commitId: commit.id,
+      path: input.path,
+    })
```

Comment:
The finding and saving should be all inside `createNewDocument`. The less code we put inside actions the better